### ISSUE TITLE
Quantization depends on Backend::isOpSupported()

### ISSUE THREAD

**Issue by jfix71** [pytorch/glow#4112]
Right now the quantization pass queries the backend to determine whether a node should be quantized. This leads to some annoyances -- namely that `Backend::isOpSupported()` is forced to use `NodeInfo` instead of just the `Node` itself. This also means that `transformForPrecisionMode()` must use backend.

To overcome the above issues, I'd like to move quantization to instead quantize everything regardless of a target backend. We could add a pass to unquantize Nodes that are unsupported by a backend later on. Then we can eventually move `Backend::isOpSupported()` to take a `Node` instead of `NodeInfo`, allowing for the backend to see what ops are before/after it, and members of the Nodes as well.

**Comment by tlepley-cadence** [comment]
@jfix71 Do you remember why `NodeInfo` was created in an alternative to passing directly `Node` to the `Backend::isOpSupported()` method ?

**Comment by jfix71** [comment]
@tlepley-cadence Yeah, the way the quantizer works (see code snippet below) is by first checking if the Node is supported (`canConvert()`) via only its types passed to `isOpSupported()`. If supported, only then does it perform quantization of inputs/dequantization of outputs and quantize the Node itself, including changing the users in the graph.

https://github.com/pytorch/glow/blob/24df161c4d0eb716473eac43673b423777a9ab48/lib/Converter/FunctionConverter.cpp#L190-L201

An alternative considered but rejected at the time was to create a quantized version of the Node along with the inputs and outputs changed, then check support by backend for this quantized node, and then if no support exists back out and swap the old node back in. Perhaps this would have been a better decision. I'm proposing a third way above, where we just simply do all of the conversion regardless of the backend, and then after the fact unquantize anything unsupported. WDYT?
